### ISSUE TITLE
Investigate whether parallel searching via different match window sizes wins performance.

### ISSUE THREAD

**Issue by sageserpent-open** [sageserpent-open/kineticMerge#134]
This is a follow on from #35.

The binary search through match window sizes is efficient, and when it moves *upwards* in size due to having found suboptimal matches, the size estimation process is more often than not bang on the money, leading directly to the optimal match.

Where things could be improved is for downward moves in size - there is a crude heuristic that guides what size to choose, but it is essentially bisecting and hoping for the best. At large window sizes, this can lead to misses, followed by further downward moves and at large window sizes this is problematic if there are lots of undiscovered optimal matches of varying size greater than the bisected size - that can confuse the size estimation process.

Would executing alternate downward moves in parallel yield any benefit?

**Comment by sageserpent-open** [comment]
Tried a spike on this; performance is *terrible*. Looking at over 4 to 5 minutes for the example merge from #35.

Running the automated test `CodeMotionAnalysisTest.matchingSectionsAreFound` alos shows a marked drop in throughput.

CPU usage is definitely up, so work is taking place in concurrent threads, but it doesn't yield any benefit whatsover.

**Comment by sageserpent-open** [comment]
Abandoning this as `wontfix`.
